### PR TITLE
[READY] Constructable Rechargers and machinery tweaks

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Rechargers now use capacitors as their stock parts and recharge faster depending on the quality of the recharger(may need balancing)
Can now also place smaller machinery on tables by clicking on the table while pulling them(microwaves and soda dispensers are some examples)

**Changelog:**
:cl:
add: Rechargers can now be crafted from science using one capacitor and a recharger board
tweak:Smaller machines like microwaves and cell chargers can now be placed on tables(Fax machines included)
tweak:can screwdriver open rechargers and then apply a crowbar to deconstruct them
/:cl:

### CODE REVIEW

**Comment by Citinited** [comment]
Nice, no more having to deconstruct and reconstruct microwaves in order to place them back on the table!

**Comment by TDSSS** [comment]
Woo, moving fax machines will be great.

**Comment by Bram961** [comment]
Constructable rechargers could influence balance. Perticularly on blob rounds where they serve as a limit to the amount of laser fire.

**Comment by Couls** [comment]
Perhaps it’ll encourage the use of reflective walls more

**Comment by Denthamos** [comment]
@Bram961 Reflective tendrils were added for a reason and are highly effective at what they do.  Blob has been really strong since they were added and nearly every time requires admins to step in with ERT/gamma now so maybe this will help bring the balance back to more of a middle ground.

**Comment by Bram961** [comment]
> @Bram961 Reflective tendrils were added for a reason -snip-

I know the PR has been closed, I was aware but the fact that it does something there would maybe necessitate a balance tag. Hence why I stated it.

File: code/game/machinery/recharger.dm [hunk 1]
```diff
@@ -6,14 +6,27 @@
...
+/obj/machinery/recharger/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/recharger(null)
+	component_parts += new /obj/item/stock_parts/capacitor(null)
+	RefreshParts()
+
+/obj/machinery/recharger/RefreshParts()
+	for(var/obj/item/stock_parts/capacitor/C in component_parts)
+		recharge_coeff = C.rating
```
Comment (variableundefined) [inline code/game/machinery/recharger.dm]: If recharge coefficient increase by rating...

File: code/game/machinery/recharger.dm [hunk 2]
```diff
@@ -88,9 +109,10 @@
 			var/obj/item/gun/energy/E = charging
 			if(E.power_supply.charge < E.power_supply.maxcharge)
 				E.power_supply.give(E.power_supply.chargerate)
-				use_power(250)
+				use_power(250 * recharge_coeff)
```
Comment (variableundefined) [inline code/game/machinery/recharger.dm]: ...And rating of capacitor increases the better they are...

Wouldn't this line results in the recharge using up more energy?

File: code/game/machinery/recharger.dm [hunk 3]
```diff
@@ -88,9 +109,10 @@
 			var/obj/item/gun/energy/E = charging
 			if(E.power_supply.charge < E.power_supply.maxcharge)
 				E.power_supply.give(E.power_supply.chargerate)
```
Comment (variableundefined) [inline code/game/machinery/recharger.dm]: Shouldn't this line be modified too for increased recharge rate? Else the recharge just get more and more wasteful for no reason at all.

**Comment by variableundefined** [comment]
PR's approved, just need some fixes

File: code/game/machinery/recharger.dm (hunk 2 above)
Comment (Couls) [reply code/game/machinery/recharger.dm]: very smooth by me, I multiplied the coefficient on the wrong thing.

File: code/game/machinery/recharger.dm (hunk 3 above)
Comment (Couls) [reply code/game/machinery/recharger.dm]: Yeah this is where the multiplication should go

**Pull request merged**